packages/core/src/sync-store/sqlite/store.ts

```diff
@@ -1373,21 +1665,22 @@ export class SqliteSyncStore implements SyncStore {
     }, {});
 
     // We can assume that source won't be empty.
-    const logOrFactorySources = sources.filter(
-      (s): s is LogSource | FactorySource =>
-        sourceIsLog(s) || sourceIsFactory(s),
+    const logSources = sources.filter(
+      (s): s is LogSource | FactoryLogSource =>
+        sourceIsLog(s) || sourceIsFactoryLog(s),
+    );
+    const callTraceSources = sources.filter(
+      (s): s is CallTraceSource | FactoryCallTraceSource =>
+        sourceIsCallTrace(s) || sourceIsFactoryCallTrace(s),
     );
     const blockSources = sources.filter(sourceIsBlock);
-    const callTraceSources = sources.filter(sourceIsCallTrace);
 
-    const shouldJoinLogs = logOrFactorySources.length !== 0;
+    const shouldJoinLogs = logSources.length !== 0;
     const shouldJoinTransactions =
-      logOrFactorySources.length !== 0 || callTraceSources.length !== 0;
+      logSources.length !== 0 || callTraceSources.length !== 0;
     const shouldJoinTraces = callTraceSources.length !== 0;
     const shouldJoinReceipts =
-      logOrFactorySources.some(
-        (source) => source.criteria.includeTransactionReceipts,
-      ) ||
+      logSources.some((source) => source.criteria.includeTransactionReceipts) ||
```

Comment:
`shouldJoinReceipts` will always be true if there is a call trace source, right? Because `callTraceSource.includeTransactionReceipts` is hard-coded to `true`?